src/core/vector/qgsvectorlayer.cpp

```diff
@@ -638,24 +638,35 @@ void QgsVectorLayer::selectByIds( const QgsFeatureIds &ids, Qgis::SelectBehavior
 {
   QGIS_PROTECT_QOBJECT_THREAD_ACCESS
 
+  // Filter out non-existent feature IDs to ensure only valid features are selected
+  QgsFeatureIds validIds;
```

Comment:
This would trigger a request on data and would have performance implication.

IMHO, we should let the way it is and take care the passed ids are existing before. Maybe we could be more explicit in the method documentation that this methods doesn't check if the ids are valid